lib/elements/dom-if.js

```diff
@@ -386,7 +385,8 @@ class DomIfFast extends DomIfBase {
     // Install runEffects hook that prevents running property effects
     // (and any nested template effects) when the `if` is false
     templateInfo.runEffects = (runEffects, changedProps, hasPaths) => {
-      const syncInfo = this.__syncInfo;
+      const instance = this.__instance;
+      let syncInfo = instance && instance.__syncInfo;
```

Comment:
The non-fast case stores `__invalidProps` on the dom-if. Why is that ok if this must be changed?

Reply:
Yeah, for the `restamp` case, it should probably be on the instance as well.  This change isn't exactly _required_ for correctness once the other part of the bug (not storing paths) is fixed; it really just prevented `restamp` from being a good workaround for the bug.  But probably good to move it.

Reply:
Ended up nulling __syncInfo at teardown, to match what we do with __invalidProps